src/Platform.Handlers/src/Xamarin.Platform.Handlers/Layouts/LayoutExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Xamarin.Platform.Layouts
+{
+	public static class LayoutExtensions
+	{
+		public static Size ComputeDesiredSize(this IFrameworkElement frameworkElement, double widthConstraint, double heightConstraint) 
+		{
+			_ = frameworkElement ?? throw new ArgumentNullException(nameof(frameworkElement));
+
+			if (frameworkElement.Handler == null)
+			{
+				return Size.Zero;
+			}
+
+			var margin = frameworkElement.Margin;
+
+			// Adjust the constraints to account for the margins
+			widthConstraint -= margin.HorizontalThickness;
+			heightConstraint -= margin.VerticalThickness;
+
+			// Determine whether the external constraints or the requested size values will determine the measurements
+			widthConstraint = LayoutManager.ResolveConstraints(widthConstraint, frameworkElement.Width);
+			heightConstraint = LayoutManager.ResolveConstraints(heightConstraint, frameworkElement.Height);
+
+			// Ask the handler to do the actual measuring								
+			var measureWithMargins = frameworkElement.Handler.GetDesiredSize(widthConstraint, heightConstraint);
+
+			// Account for the margins when reporting the desired size value
+			return new Size(measureWithMargins.Width + margin.HorizontalThickness,
+				measureWithMargins.Height + margin.VerticalThickness);
+		}
+
+		public static Rectangle ComputeFrame(this IFrameworkElement frameworkElement, Rectangle bounds) 
+		{
+			var margin = frameworkElement.Margin;
+
+			// If the margins are too big for the bounds, then simply collapse them to zero
+			var frameWidth = Math.Max(0, bounds.Width - margin.HorizontalThickness);
+			var frameHeight = Math.Max(0, bounds.Height - margin.VerticalThickness);
+
+			return new Rectangle(bounds.X + margin.Left, bounds.Y + margin.Top, 
```

Comment:
I'm sorry I'm being picky about this 😅 but accounting for RTL #3066 is really easy, it is a one line fix #11469 . No tests will fail if you do this because all of the current margin tests (even RTL ones) use equal margins on both sides so the issue doesn't appear.

Reply:
Yes, this will be in a future PR.